### ISSUE TITLE
Unwarranted "Bad number of parameters" when calling a function with a constant on a subobject (containing an array of at least 5 bytes) that shares its name with a constant in another subobject that the computation of the constant parameter depends on.

### ISSUE THREAD

**Issue by Wuerfel21** [totalspectrum/spin2cpp#127]
Wow that's a specific one, isn't it. I'll better let the code speak for itself:

`flexspin_badparcnt.spin2`:
```spin
OBJ
subobj: "flexspin_badparcnt_subobj.spin2"
conobj: "flexspin_badparcnt_con.spin2"

PUB main()

  subobj.myfun(conobj.FUNNY)

```

`flexspin_badparcnt_subobj.spin2`:
```spin
VAR byte lol[5]
PUB myfun(a1)
```

`flexspin_badparcnt_con.spin2`:
```spin
CON
SUBOBJ = $1000
FUNNY = SUBOBJ + 8*2
PUB nopub()
```

This all errs with `Bad number of parameters in call to myfun: expected 1 found 2`

And yes, I did check all those conditions in the title.

Also perhaps helpful: Version `5.2.1-beta-da3db25` handles this situation correctly. (I am currently on `5.4.2-beta-243b8a5`)

**Comment by totalspectrum** [comment]
That was a good one :). One part of the code was using type information generated based on the wrong object context (the top level object instead of the subobject). Normally this worked, but if the top level object had an object name that happened to match a name used in calculating the constant then the type information from the top level object was used. In really old versions of flexspin this didn't matter, the type information got thrown away, but in order to fix some bugs that had to stop happening... but then this situation arose (an object with more than 5 bytes needs two parameters to pass it... this isn't really relevant for Spin2, which shouldn't be able to pass objects directly, but the bug made it think we were trying to pass the object).

tl;dr : it's fixed now in github, thank you
